app/controllers/users.js

```diff
@@ -1,26 +1,60 @@
 const util = require('util');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
 
 const User = require('../models').users;
 const logger = require('../logger');
 const error = require('../errors');
-const { hashPassword } = require('../services/users');
+const { hashPassword, findByEmail } = require('../services/users');
 
-const { userCreationError } = error;
+const { userCreationError, userLoginError } = error;
 
-exports.postUser = (req, res, next) => {
+exports.signUpUser = (req, res, next) => {
   const { first_name: firstName, last_name: lastName, email, password } = req.body;
-  return hashPassword(password)
-    .then(() =>
-      User.create({ firstName, lastName, email, password })
-        .then(user => {
-          logger.info(`Usuario creado para: ${firstName} ${lastName}`);
-          return user;
+  return findByEmail(email)
+    .then(foundUser => {
+      if (foundUser) {
+        throw userCreationError(`User for ${email} already exists!`);
+      } else {
+        return hashPassword(password)
+          .then(hashedPassword =>
+            User.create({ firstName, lastName, email, password: hashedPassword })
+              .then(user => {
+                logger.info(`New user created for: ${firstName} ${lastName}`);
+                return user;
+              })
+              .catch(err => {
+                logger.error(util.inspect(err));
+                throw userCreationError(err.message);
+              })
+          )
+          .then(response => res.status(201).send(response))
+          .catch(next);
+      }
+    })
+    .catch(next);
+};
+
+exports.signInUser = (req, res, next) =>
+  findByEmail(req.body.email)
+    .then(user =>
+      bcrypt
+        .compare(req.body.password, user.password)
+        .then(result => {
+          if (result) {
+            const token = jwt.sign(
+              { id: user.id, name: [user.firstName, user.lastName].join(' ') },
+              process.env.TOKEN_SECRET
+            );
+            res.header('auth-token', token);
```

Comment:
This is not necessary. In node, we send the token and that it's all

Reply:
:ok_hand: